Source/SPTablesList.m

```diff
@@ -147,10 +148,32 @@ - (void)awakeFromNib
 	
 	[tablesListView registerForDraggedTypes:@[SPNavigatorTableDataPasteboardDragType]];
 
+	[tablesListMenu setDelegate:self];
+
 	//create the charset helper
 	addTableCharsetHelper = [[SPCharsetCollationHelper alloc] initWithCharsetButton:tableEncodingButton CollationButton:tableCollationButton];
 }
 
+/**
+ * Not a menuDidOpen because it fires after tableViewSelectionDidChange which is responsible for DB View (and other things) refresh
+ */
+- (void)rightMouseDown:(NSEvent *)event
+{
+	isMenuOpened = true;
+}
+
+- (void)menuDidClose:(NSMenu *)menu {
+	isMenuOpened = false;
+}
```

Comment:
Looks like nothing calls this?

Reply:
It's called automatically via NSMenuDelegate